workspaces/scorecard/plugins/scorecard-backend-module-jira/src/clients/base.ts

```diff
@@ -0,0 +1,184 @@
+      product: jiraConfig.getString(CONFIG_PRODUCT) as Product,
+      apiVersion: jiraConfig.getOptionalString(CONFIG_API_VERSION),
+    };
+
+    const jiraOptions = config.getOptionalConfig(JIRA_OPTIONS_PATH);
+    if (jiraOptions) {
+      this.options = {
+        mandatoryFilter: jiraOptions.getOptionalString(CONFIG_MANDATORY_FILTER),
+        customFilter: jiraOptions.getOptionalString(CONFIG_CUSTOM_FILTER),
+      };
+    }
+  }
+
+  protected async sendRequest({
+    url,
+    method,
+    headers = {},
+    body = '',
+  }: RequestOptions): Promise<unknown> {
+    try {
+      const response = await fetch(url, {
+        method,
+        headers: {
+          Accept: 'application/json',
+          'Content-Type': 'application/json',
+          'X-Atlassian-Token': 'no-check',
+          ...headers,
+        },
+        body,
+      });
+
+      if (!response.ok) {
+        throw new Error(`Jira request failed with status ${response.status}`);
+      }
+
+      return response.json();
+    } catch (error) {
+      throw new Error(
+        `Jira error message: ${
+          error instanceof Error ? error.message : String(error)
+        }`,
+      );
+    }
+  }
+
+  protected getFiltersFromEntity(entity: Entity): JiraEntityFilters {
+    const annotations = entity?.metadata?.annotations || {};
+
+    const projectKey = annotations[ANNOTATION_JIRA_PROJECT_KEY];
+    if (!projectKey) {
+      throw new Error(
+        `Missing required 'jira/project-key' annotation for entity '${
+          entity.metadata?.name || 'unknown'
+        }'`,
+      );
+    }
+
+    const filters: JiraEntityFilters = { project: `project = "${projectKey}"` };
+
+    const component = annotations[ANNOTATION_JIRA_COMPONENT];
+    if (component) filters.component = `component = "${component}"`;
+
+    const label = annotations[ANNOTATION_JIRA_LABEL];
+    if (label) filters.label = `labels = "${label}"`;
+
+    const team = annotations[ANNOTATION_JIRA_TEAM];
+    if (team) filters.team = `team = ${team}`;
+
+    const customFilter = annotations[ANNOTATION_JIRA_CUSTOM_FILTER];
+    if (customFilter) filters.customFilter = customFilter;
+
+    return filters;
+  }
+
+  protected buildJqlFilters(filters: JiraEntityFilters): string {
+    const { customFilter: annotationCustomFilter } = filters;
+    const { mandatoryFilter, customFilter: optionsCustomFilter } =
+      this.options || {};
+
+    const defaultFilterQuery = mandatoryFilter ?? JIRA_MANDATORY_FILTER;
+
+    const customFilterQuery =
+      !annotationCustomFilter && optionsCustomFilter
+        ? optionsCustomFilter
+        : null;
+
+    return Object.values({
+      defaultFilterQuery,
+      customFilterQuery,
+      ...filters,
+    })
+      .filter(value => value !== null && value !== undefined)
+      .map(value => {
+        if (typeof value === 'object') return JSON.stringify(value);
+        return value.toString();
+      })
+      .join(' AND ');
```

Comment:
Please add () around the separate filter parts to be sure that they are not mixed.

(Also if I know that most programming and I think also JQL says that an AND is more important then an OR.